applications/RANSApplication/custom_strategies/algebraic_flux_corrected_steady_scalar_scheme.h

```diff
@@ -0,0 +1,582 @@
+                    r_node_1.SetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX, p_plus);
+                    r_node_1.SetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, q_plus);
+                    r_node_1.SetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX, p_minus);
+                    r_node_1.SetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, q_minus);
+                    r_node_1.UnSetLock();
+                }
+            });
+        }
+
+        Communicator& r_communicator = rModelPart.GetCommunicator();
+        r_communicator.AssembleNonHistoricalData(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX);
+        r_communicator.AssembleNonHistoricalData(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
+        r_communicator.AssembleNonHistoricalData(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX);
+        r_communicator.AssembleNonHistoricalData(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
+
+        KRATOS_CATCH("")
+    }
+
+    void Update(
+        ModelPart& rModelPart,
+        DofsArrayType& rDofSet,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        mpDofUpdater->UpdateDofs(rDofSet, rDx);
+
+        KRATOS_CATCH("");
+    }
+
+    void Clear() override
+    {
+        this->mpDofUpdater->Clear();
+    }
+
+    void CalculateSystemContributions(
+        Element& rElement,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        Element::EquationIdVectorType& rEquationIdVector,
+        const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const auto k = OpenMPUtils::ThisThread();
+
+        this->CalculateSystemMatrix<Element>(rElement, rLHS_Contribution, rRHS_Contribution,
+                                             mAuxMatrix[k], rCurrentProcessInfo);
+        rElement.EquationIdVector(rEquationIdVector, rCurrentProcessInfo);
+
+        this->CalculateArtificialDiffusionMatrix(mAuxMatrix[k], rLHS_Contribution);
+
+        AddAntiDiffusiveFluxes(rRHS_Contribution, rLHS_Contribution, rElement,
+                               mAuxMatrix[k]);
+        noalias(rLHS_Contribution) += mAuxMatrix[k];
+
+        rElement.GetValuesVector(mValues[k]);
+        noalias(rRHS_Contribution) -= prod(rLHS_Contribution, mValues[k]);
+
+        KRATOS_CATCH("");
+    }
+
+    void CalculateSystemContributions(
+        Condition& rCondition,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        Element::EquationIdVectorType& rEquationIdVector,
+        const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const auto k = OpenMPUtils::ThisThread();
+
+        this->CalculateSystemMatrix<Condition>(rCondition, rLHS_Contribution, rRHS_Contribution,
+                                               mAuxMatrix[k], rCurrentProcessInfo);
+        rCondition.EquationIdVector(rEquationIdVector, rCurrentProcessInfo);
+
```

Comment:
Shouldn't the RHS contributions be added too?

Reply:
It is added via templates `CalculateSystemMatrix` method.